ls1intum/Artemis
# `Development`: Remove unnecessary server call to get notification settings

### Checklist
#### General
- [x] This is a small issue that I tested on a test server.
#### Server
- not changed
#### Client

### Motivation and Context

The notification sidebar loads the notification setting at start up to be able to handle group notifications (due to multicast groups, i.e. filters do not work on the server side here). If you open the notification settings one aditional server call is performed to fetch the notification setting. @krusche told me to change this so the notification settings page reuses the already available settings from the sidebar.


### Description
To avoid the direct coupling between components I added a variable to the notification setting service which holds the newest/latest version of the settings with getter and setter methods. I.e. when the sidebar loads the settings its sets them in the service and when the settings page is opened it checks if there are settings in the service. If not it loads them from the server as before. This usecase is still needed in the rare case when someone reloads the settings page itself.


### Steps for Testing
Log into Artemis and open your browser developer debugger tool and select the network panel.
Open the settings and check if they are visible without any loading-lag (instantly).
Check in the network console that no redundant notification-settings are fetched when switching from the homepage to the notification settings page. 

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage
The biggest change beside a getter and setter was added to the notification-settings.component -> I added a new test case -> coverage > 85%
